ui/app/AppLayouts/Communities/panels/MintTokensSettingsPanel.qml

```diff
@@ -257,265 +258,275 @@ StackView {
+            readonly property alias tokenName: view.token.name
 
-                width: 521
-                title: qsTr("Delete %1").arg(tokenName)
-                acceptBtnText: qsTr("Delete %1 token").arg(tokenName)
-                alertText: qsTr("%1 is not yet minted, are you sure you want to delete it? All data associated with this token including its icon and description will be permanently deleted.").arg(tokenName)
+            width: 521
+            title: qsTr("Delete %1").arg(tokenName)
+            acceptBtnText: qsTr("Delete %1 token").arg(tokenName)
+            alertText: qsTr("%1 is not yet minted, are you sure you want to delete it? All data associated with this token including its icon and description will be permanently deleted.").arg(tokenName)
 
-                onAcceptClicked: {
-                    root.deleteToken(tokenViewPage.tokenKey)
-                    root.navigateBack()
+            onAcceptClicked: {
+                root.deleteToken(tokenViewPage.tokenKey)
+                root.navigateBack()
+            }
+            onCancelClicked: close()
+        }
+    }
+
+    Component {
+        id: tokenViewComponent
+
+        Item {
+            id: tokenViewPageWrapper
+
+            property string tokenKey
+
+            Repeater {
+                model: SortFilterProxyModel {
+                    sourceModel: root.tokensModel
+                    filters: ValueFilter {
+                        roleName: "contractUniqueKey"
+                        value: tokenViewPageWrapper.tokenKey
+                    }
+                }
+
+                delegate: TokenViewPage {
+                    implicitWidth: 0
+                    anchors.fill: parent
+
+                    tokenOwnersModel: model.tokenOwnersModel
+                    airdropKey: model.symbol // TO BE REMOVED: When airdrop backend is ready to use token key instead of symbol
+
+                    token.accountName: model.accountName
+                    token.artworkSource: model.image
+                    token.chainIcon: model.chainIcon
+                    token.chainId: model.chainId
+                    token.chainName: model.chainName
+                    token.decimals: model.decimals
+                    token.deployState: model.deployState
+                    token.description: model.description
+                    token.infiniteSupply: model.infiniteSupply
+                    token.key: model.contractUniqueKey
+                    token.name: model.name
+                    token.remainingTokens: model.remainingSupply
+                    token.remotelyDestruct: model.remoteSelfDestruct
+                    token.supply: model.supply
+                    token.symbol: model.symbol
+                    token.transferable: model.transferable
+                    token.type: model.tokenType
+
+                    // TODO: Backend
+                    //token.accountAddress: model.accountAddress
```

Comment:
Leftover?

Reply:
No, those properties are in-deed not provided by the backend right now but should be.

Reply:
:disappointed:

Reply:
Oh true, I misread  it and was thinking on `accountName`!